stonesoup/models/measurement/nonlinear.py

```diff
@@ -782,12 +782,200 @@ def function(self, state, noise=False, **kwargs) -> StateVector:
         bearings = [Bearing(i) for i in phi]
 
         return StateVectors([bearings, rho, rr]) + noise
+    
+    def inverse_function(self, detection, **kwargs) -> StateVector:
+        phi, rho, rho_rate = detection.state_vector
+
+        x, y = pol2cart(rho, phi)
+        # because only rho_rate is known, only the components in
+        # x,y of the range rate can be found.
+        x_rate = np.cos(phi) * rho_rate
+        y_rate = np.sin(phi) * rho_rate
+
+        inv_rotation_matrix = inv(self.rotation_matrix)
+        out_vector = StateVector([[0.], [0.], [0.], [0.], [0.], [0.]])
+        out_vector[self.mapping, 0] = x, y
+        out_vector[self.velocity_mapping, 0] = x_rate, y_rate
```

Comment:
From test failures, these mappings are 3D, but this is only assigning 2D values.

Reply:
This could be fixed by setting z and z_rate to 0 when mapping out_vector - is this supported by the theory, or is there a way to calculate the z parameters from measurements of range/bearing/range rate?